autokeras/task.py

```diff
@@ -167,3 +167,176 @@ def __init__(self,
             max_trials=max_trials,
             directory=directory,
             seed=seed)
+
+
+class SupervisedStructuredDataPipeline(auto_model.AutoModel):
+
+    def __init__(self, outputs, column_names, column_types, **kwargs):
+        # TODO: support customized column_types.
+        inputs = node.StructuredDataInput()
+        inputs.column_types = column_types
+        inputs.column_names = column_names
+        if column_types:
+            for column_type in column_types.values():
+                if column_type not in ['categorical', 'numerical']:
+                    raise ValueError(
+                        'Column_types should be either "categorical" '
+                        'or "numerical", but got {name}'.format(name=column_type))
+        if column_names and column_types:
+            for column_name in column_types:
+                if column_name not in column_names:
+                    raise ValueError('Column_names and column_types are '
+                                     'mismatched. Cannot find column name '
+                                     '{name} in the data.'.format(name=column_name))
+        super().__init__(inputs=inputs,
+                         outputs=outputs,
+                         **kwargs)
+
+    def fit(self,
+            x=None,
+            y=None,
+            validation_split=0,
+            validation_data=None,
+            **kwargs):
+        """Search for the best model and hyperparameters for the task.
+
+        # Arguments
+            x: String, numpy.ndarray, pandas.DataFrame or tensorflow.Dataset.
+                Training data x. If the data is from a csv file, it should be a
+                string specifying the path of the csv file of the training data.
+            y: String, numpy.ndarray, or tensorflow.Dataset. Training data y.
```

Comment:
- How is it possible for `y` to be a dataset?
- Wouldn't "string" also be valid if `x` is a dataframe? (y = name of target column)

It seems to me the possibilities are:

- x = dataset, y = None
- x = path to csv file, y = column name
- x = dataframe, y = column name
- x = np, y = np

Right?

Reply:
1. If x and y are both datasets, we would zip them together.
2. If x is a dataframe, we are not supporting the column name string of y.
3. Those are the typical ways of passing x and y. The user can also do something like: x=dataframe, y = np.